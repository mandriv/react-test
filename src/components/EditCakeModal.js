import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Modal, Form, Button, Icon } from 'semantic-ui-react';

/*
  EditCakeModal description
*/

export default class EditCakeModal extends Component {

  constructor(props) {
    super(props);
    const { cake } = props;
    this.state = {
      title: cake.title,
      desc: cake.desc,
      image: cake.image,
      id: cake.id, // eslint-disable-line
    };
  }

  render() {
    const { open, closeIcon, closeOnDocumentClick, onClose } = this.props;
    const { title, desc, image } = this.state;
    return (
      <Modal
        open={open}
        closeIcon={closeIcon}
        closeOnDocumentClick={closeOnDocumentClick}
        onClose={onClose}
      >
        <Modal.Header>Edit cake</Modal.Header>
        <Modal.Content>
          <Form>
            <Form.Field>
              <label htmlFor="cakeTitle">
                Title
                <Form.Input
                  placeholder="Cake's title"
                  value={title}
                  onChange={(ev, data) => this.setState({ title: data.value })}
                  id="cakeTitle"
                />
              </label>
            </Form.Field>
            <Form.Field>
              <label htmlFor="cakeDescription">
                Description
                <Form.TextArea
                  placeholder="Cake's description"
                  value={desc}
                  onChange={(ev, data) => this.setState({ desc: data.value })}
                  id="cakeDescription"
                />
              </label>
            </Form.Field>
            <Form.Field>
              <label htmlFor="cakeImg">
                Image URL
                <Form.Input
                  placeholder="Cake's image URL"
                  value={image}
                  onChange={(ev, data) => this.setState({ image: data.value })}
                  id="cakeImg"
                />
              </label>
            </Form.Field>
          </Form>
        </Modal.Content>
        <Modal.Actions>
          <Button
            icon
            negative
            onClick={() => this.props.onClose()}
          >
            <Icon name="remove" />
            Cancel
          </Button>
          <Button
            icon
            positive
            onClick={() => {
              this.props.onSubmit(this.state);
              this.props.onClose();
            }}
          >
            <Icon name="checkmark" />
            Save
          </Button>
        </Modal.Actions>
      </Modal>
    );
  }

}

EditCakeModal.propTypes = {
  cake: PropTypes.object.isRequired,
  onSubmit: PropTypes.func,
  onClose: PropTypes.func,
  closeOnDocumentClick: PropTypes.bool,
  closeIcon: PropTypes.bool,
  open: PropTypes.bool,
};

EditCakeModal.defaultProps = {
  onSubmit: () => null,
  onClose: () => null,
  closeOnDocumentClick: false,
  closeIcon: false,
  open: false,
};
