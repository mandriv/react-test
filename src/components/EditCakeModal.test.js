import React from 'react';
import { shallow } from 'enzyme';
import EditCakeModal from './EditCakeModal';

describe('Component: <EditCakeModal>', () => {
  it('Renders <EditCakeModal>', () => {
    const cake = {
      title: 'Testy Cake',
      desc: 'description of test cake',
      image: 'https://picsum.photos/400',
    };
    const editCakeComponent = shallow(<EditCakeModal cake={cake} />);
    expect(editCakeComponent).toMatchSnapshot();
  });
});
