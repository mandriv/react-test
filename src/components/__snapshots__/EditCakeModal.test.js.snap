// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <EditCakeModal> Renders <EditCakeModal> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditCakeModal
    cake={
        Object {
            "desc": "description of test cake",
            "image": "https://picsum.photos/400",
            "title": "Testy Cake",
          }
    }
    closeIcon={false}
    closeOnDocumentClick={false}
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ModalHeader>
          Edit cake
</ModalHeader>,
        <ModalContent>
          <Form
                    as="form"
          >
                    <FormField>
                              <label
                                        htmlFor="cakeTitle"
                              >
                                        Title
                                        <FormInput
                                                  as={[Function]}
                                                  control={[Function]}
                                                  id="cakeTitle"
                                                  onChange={[Function]}
                                                  placeholder="Cake's title"
                                                  value="Testy Cake"
                                        />
                              </label>
                    </FormField>
                    <FormField>
                              <label
                                        htmlFor="cakeDescription"
                              >
                                        Description
                                        <FormTextArea
                                                  as={[Function]}
                                                  control={[Function]}
                                                  id="cakeDescription"
                                                  onChange={[Function]}
                                                  placeholder="Cake's description"
                                                  value="description of test cake"
                                        />
                              </label>
                    </FormField>
                    <FormField>
                              <label
                                        htmlFor="cakeImg"
                              >
                                        Image URL
                                        <FormInput
                                                  as={[Function]}
                                                  control={[Function]}
                                                  id="cakeImg"
                                                  onChange={[Function]}
                                                  placeholder="Cake's image URL"
                                                  value="https://picsum.photos/400"
                                        />
                              </label>
                    </FormField>
          </Form>
</ModalContent>,
        <ModalActions>
          <Button
                    as="button"
                    icon={true}
                    negative={true}
                    onClick={[Function]}
          >
                    <Icon
                              as="i"
                              name="remove"
                    />
                    Cancel
          </Button>
          <Button
                    as="button"
                    icon={true}
                    onClick={[Function]}
                    positive={true}
          >
                    <Icon
                              as="i"
                              name="checkmark"
                    />
                    Save
          </Button>
</ModalActions>,
      ],
      "closeIcon": false,
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "onClose": [Function],
      "open": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Edit cake",
        },
        "ref": null,
        "rendered": "Edit cake",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Form
            as="form"
>
            <FormField>
                        <label
                                    htmlFor="cakeTitle"
                        >
                                    Title
                                    <FormInput
                                                as={[Function]}
                                                control={[Function]}
                                                id="cakeTitle"
                                                onChange={[Function]}
                                                placeholder="Cake's title"
                                                value="Testy Cake"
                                    />
                        </label>
            </FormField>
            <FormField>
                        <label
                                    htmlFor="cakeDescription"
                        >
                                    Description
                                    <FormTextArea
                                                as={[Function]}
                                                control={[Function]}
                                                id="cakeDescription"
                                                onChange={[Function]}
                                                placeholder="Cake's description"
                                                value="description of test cake"
                                    />
                        </label>
            </FormField>
            <FormField>
                        <label
                                    htmlFor="cakeImg"
                        >
                                    Image URL
                                    <FormInput
                                                as={[Function]}
                                                control={[Function]}
                                                id="cakeImg"
                                                onChange={[Function]}
                                                placeholder="Cake's image URL"
                                                value="https://picsum.photos/400"
                                    />
                        </label>
            </FormField>
</Form>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "form",
            "children": Array [
              <FormField>
                <label
                                htmlFor="cakeTitle"
                >
                                Title
                                <FormInput
                                                as={[Function]}
                                                control={[Function]}
                                                id="cakeTitle"
                                                onChange={[Function]}
                                                placeholder="Cake's title"
                                                value="Testy Cake"
                                />
                </label>
</FormField>,
              <FormField>
                <label
                                htmlFor="cakeDescription"
                >
                                Description
                                <FormTextArea
                                                as={[Function]}
                                                control={[Function]}
                                                id="cakeDescription"
                                                onChange={[Function]}
                                                placeholder="Cake's description"
                                                value="description of test cake"
                                />
                </label>
</FormField>,
              <FormField>
                <label
                                htmlFor="cakeImg"
                >
                                Image URL
                                <FormInput
                                                as={[Function]}
                                                control={[Function]}
                                                id="cakeImg"
                                                onChange={[Function]}
                                                placeholder="Cake's image URL"
                                                value="https://picsum.photos/400"
                                />
                </label>
</FormField>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <label
                  htmlFor="cakeTitle"
>
                  Title
                  <FormInput
                                    as={[Function]}
                                    control={[Function]}
                                    id="cakeTitle"
                                    onChange={[Function]}
                                    placeholder="Cake's title"
                                    value="Testy Cake"
                  />
</label>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Title",
                    <FormInput
                      as={[Function]}
                      control={[Function]}
                      id="cakeTitle"
                      onChange={[Function]}
                      placeholder="Cake's title"
                      value="Testy Cake"
/>,
                  ],
                  "htmlFor": "cakeTitle",
                },
                "ref": null,
                "rendered": Array [
                  "Title",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": [Function],
                      "control": [Function],
                      "id": "cakeTitle",
                      "onChange": [Function],
                      "placeholder": "Cake's title",
                      "value": "Testy Cake",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "label",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <label
                  htmlFor="cakeDescription"
>
                  Description
                  <FormTextArea
                                    as={[Function]}
                                    control={[Function]}
                                    id="cakeDescription"
                                    onChange={[Function]}
                                    placeholder="Cake's description"
                                    value="description of test cake"
                  />
</label>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Description",
                    <FormTextArea
                      as={[Function]}
                      control={[Function]}
                      id="cakeDescription"
                      onChange={[Function]}
                      placeholder="Cake's description"
                      value="description of test cake"
/>,
                  ],
                  "htmlFor": "cakeDescription",
                },
                "ref": null,
                "rendered": Array [
                  "Description",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": [Function],
                      "control": [Function],
                      "id": "cakeDescription",
                      "onChange": [Function],
                      "placeholder": "Cake's description",
                      "value": "description of test cake",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "label",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <label
                  htmlFor="cakeImg"
>
                  Image URL
                  <FormInput
                                    as={[Function]}
                                    control={[Function]}
                                    id="cakeImg"
                                    onChange={[Function]}
                                    placeholder="Cake's image URL"
                                    value="https://picsum.photos/400"
                  />
</label>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Image URL",
                    <FormInput
                      as={[Function]}
                      control={[Function]}
                      id="cakeImg"
                      onChange={[Function]}
                      placeholder="Cake's image URL"
                      value="https://picsum.photos/400"
/>,
                  ],
                  "htmlFor": "cakeImg",
                },
                "ref": null,
                "rendered": Array [
                  "Image URL",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": [Function],
                      "control": [Function],
                      "id": "cakeImg",
                      "onChange": [Function],
                      "placeholder": "Cake's image URL",
                      "value": "https://picsum.photos/400",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "label",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Button
              as="button"
              icon={true}
              negative={true}
              onClick={[Function]}
>
              <Icon
                            as="i"
                            name="remove"
              />
              Cancel
</Button>,
            <Button
              as="button"
              icon={true}
              onClick={[Function]}
              positive={true}
>
              <Icon
                            as="i"
                            name="checkmark"
              />
              Save
</Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "children": Array [
                <Icon
                  as="i"
                  name="remove"
/>,
                "Cancel",
              ],
              "icon": true,
              "negative": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "name": "remove",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "Cancel",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "children": Array [
                <Icon
                  as="i"
                  name="checkmark"
/>,
                "Save",
              ],
              "icon": true,
              "onClick": [Function],
              "positive": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "name": "checkmark",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "Save",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ModalHeader>
            Edit cake
</ModalHeader>,
          <ModalContent>
            <Form
                        as="form"
            >
                        <FormField>
                                    <label
                                                htmlFor="cakeTitle"
                                    >
                                                Title
                                                <FormInput
                                                            as={[Function]}
                                                            control={[Function]}
                                                            id="cakeTitle"
                                                            onChange={[Function]}
                                                            placeholder="Cake's title"
                                                            value="Testy Cake"
                                                />
                                    </label>
                        </FormField>
                        <FormField>
                                    <label
                                                htmlFor="cakeDescription"
                                    >
                                                Description
                                                <FormTextArea
                                                            as={[Function]}
                                                            control={[Function]}
                                                            id="cakeDescription"
                                                            onChange={[Function]}
                                                            placeholder="Cake's description"
                                                            value="description of test cake"
                                                />
                                    </label>
                        </FormField>
                        <FormField>
                                    <label
                                                htmlFor="cakeImg"
                                    >
                                                Image URL
                                                <FormInput
                                                            as={[Function]}
                                                            control={[Function]}
                                                            id="cakeImg"
                                                            onChange={[Function]}
                                                            placeholder="Cake's image URL"
                                                            value="https://picsum.photos/400"
                                                />
                                    </label>
                        </FormField>
            </Form>
</ModalContent>,
          <ModalActions>
            <Button
                        as="button"
                        icon={true}
                        negative={true}
                        onClick={[Function]}
            >
                        <Icon
                                    as="i"
                                    name="remove"
                        />
                        Cancel
            </Button>
            <Button
                        as="button"
                        icon={true}
                        onClick={[Function]}
                        positive={true}
            >
                        <Icon
                                    as="i"
                                    name="checkmark"
                        />
                        Save
            </Button>
</ModalActions>,
        ],
        "closeIcon": false,
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "onClose": [Function],
        "open": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Edit cake",
          },
          "ref": null,
          "rendered": "Edit cake",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Form
              as="form"
>
              <FormField>
                            <label
                                          htmlFor="cakeTitle"
                            >
                                          Title
                                          <FormInput
                                                        as={[Function]}
                                                        control={[Function]}
                                                        id="cakeTitle"
                                                        onChange={[Function]}
                                                        placeholder="Cake's title"
                                                        value="Testy Cake"
                                          />
                            </label>
              </FormField>
              <FormField>
                            <label
                                          htmlFor="cakeDescription"
                            >
                                          Description
                                          <FormTextArea
                                                        as={[Function]}
                                                        control={[Function]}
                                                        id="cakeDescription"
                                                        onChange={[Function]}
                                                        placeholder="Cake's description"
                                                        value="description of test cake"
                                          />
                            </label>
              </FormField>
              <FormField>
                            <label
                                          htmlFor="cakeImg"
                            >
                                          Image URL
                                          <FormInput
                                                        as={[Function]}
                                                        control={[Function]}
                                                        id="cakeImg"
                                                        onChange={[Function]}
                                                        placeholder="Cake's image URL"
                                                        value="https://picsum.photos/400"
                                          />
                            </label>
              </FormField>
</Form>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "form",
              "children": Array [
                <FormField>
                  <label
                                    htmlFor="cakeTitle"
                  >
                                    Title
                                    <FormInput
                                                      as={[Function]}
                                                      control={[Function]}
                                                      id="cakeTitle"
                                                      onChange={[Function]}
                                                      placeholder="Cake's title"
                                                      value="Testy Cake"
                                    />
                  </label>
</FormField>,
                <FormField>
                  <label
                                    htmlFor="cakeDescription"
                  >
                                    Description
                                    <FormTextArea
                                                      as={[Function]}
                                                      control={[Function]}
                                                      id="cakeDescription"
                                                      onChange={[Function]}
                                                      placeholder="Cake's description"
                                                      value="description of test cake"
                                    />
                  </label>
</FormField>,
                <FormField>
                  <label
                                    htmlFor="cakeImg"
                  >
                                    Image URL
                                    <FormInput
                                                      as={[Function]}
                                                      control={[Function]}
                                                      id="cakeImg"
                                                      onChange={[Function]}
                                                      placeholder="Cake's image URL"
                                                      value="https://picsum.photos/400"
                                    />
                  </label>
</FormField>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <label
                    htmlFor="cakeTitle"
>
                    Title
                    <FormInput
                                        as={[Function]}
                                        control={[Function]}
                                        id="cakeTitle"
                                        onChange={[Function]}
                                        placeholder="Cake's title"
                                        value="Testy Cake"
                    />
</label>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Title",
                      <FormInput
                        as={[Function]}
                        control={[Function]}
                        id="cakeTitle"
                        onChange={[Function]}
                        placeholder="Cake's title"
                        value="Testy Cake"
/>,
                    ],
                    "htmlFor": "cakeTitle",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Title",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "as": [Function],
                        "control": [Function],
                        "id": "cakeTitle",
                        "onChange": [Function],
                        "placeholder": "Cake's title",
                        "value": "Testy Cake",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "label",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <label
                    htmlFor="cakeDescription"
>
                    Description
                    <FormTextArea
                                        as={[Function]}
                                        control={[Function]}
                                        id="cakeDescription"
                                        onChange={[Function]}
                                        placeholder="Cake's description"
                                        value="description of test cake"
                    />
</label>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Description",
                      <FormTextArea
                        as={[Function]}
                        control={[Function]}
                        id="cakeDescription"
                        onChange={[Function]}
                        placeholder="Cake's description"
                        value="description of test cake"
/>,
                    ],
                    "htmlFor": "cakeDescription",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Description",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "as": [Function],
                        "control": [Function],
                        "id": "cakeDescription",
                        "onChange": [Function],
                        "placeholder": "Cake's description",
                        "value": "description of test cake",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "label",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <label
                    htmlFor="cakeImg"
>
                    Image URL
                    <FormInput
                                        as={[Function]}
                                        control={[Function]}
                                        id="cakeImg"
                                        onChange={[Function]}
                                        placeholder="Cake's image URL"
                                        value="https://picsum.photos/400"
                    />
</label>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Image URL",
                      <FormInput
                        as={[Function]}
                        control={[Function]}
                        id="cakeImg"
                        onChange={[Function]}
                        placeholder="Cake's image URL"
                        value="https://picsum.photos/400"
/>,
                    ],
                    "htmlFor": "cakeImg",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Image URL",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "as": [Function],
                        "control": [Function],
                        "id": "cakeImg",
                        "onChange": [Function],
                        "placeholder": "Cake's image URL",
                        "value": "https://picsum.photos/400",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "label",
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Button
                as="button"
                icon={true}
                negative={true}
                onClick={[Function]}
>
                <Icon
                                as="i"
                                name="remove"
                />
                Cancel
</Button>,
              <Button
                as="button"
                icon={true}
                onClick={[Function]}
                positive={true}
>
                <Icon
                                as="i"
                                name="checkmark"
                />
                Save
</Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "children": Array [
                  <Icon
                    as="i"
                    name="remove"
/>,
                  "Cancel",
                ],
                "icon": true,
                "negative": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "i",
                    "name": "remove",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Cancel",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "children": Array [
                  <Icon
                    as="i"
                    name="checkmark"
/>,
                  "Save",
                ],
                "icon": true,
                "onClick": [Function],
                "positive": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "i",
                    "name": "checkmark",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Save",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
